doc/feature/spikeReport.md

```diff
@@ -0,0 +1,289 @@
+         *        understood as a filter: any included GID which is not
+         *        actually part of the report will be silently ignored.
+         */
+        SpikeReport( const URI& uri, const GIDSet& subset );
+
+        /**
+         * Release all resources.
+         * Any threads waiting on futures will return immediately and calling
+         * get will throw a runtime_exception.
+         */
+        ~SPikeReport();
+
+        const URI& getURI();
+
+        /** ? */
+        float getStartTime() const;
+        float getEndTime() const;
+
+        /**
+         * @return the end time of the latest complete read/write operation or
+         *         UNDEFINED_TIMESTAMP if no operation has been issued.
+         *
+         * Operations are deemed as complete when the returned future is ready.
+         *
+         * The time inverval to which getCurrentTime refers is open on the
+         * right. That means that no read of write operation may return or take
+         * a timestamp < getCurrentTime().
+         */
+        float getCurrentTime() const;
+
+        /**
+         * Read spikes until getCurrentTime becomes > min, the end of the source
+         * is reached or closed.
+         *
+         * Preconditions:
+         * - r.getState() is 0 or ENDED.
+         * - The report was open in read mode.
+         * - There is no previous read operation with a pending future.
+         * - min >= getCurrentTime() or UNDEFINED_TIMESTAMP.
+         * Throw std::logic_error if one of the preconditions is not fulfilled.
+         *
+         * Postconditions:
+         * Let:
+         *  - r be the SpikeReport
+         *  - f be the returned future by r.read(min)
+         *  - and s = r.getCurrentTime() before read is called or -inf if
+         *    undefined:
+         * After f.wait() returns the following are all true:
+         *  - r.getCurrentTime() >= s
+         *  - If r.getState() == 0, then r.getCurrentTime() > min
+         *  - For each spike timestamp t_s in f.get(),
+         *    s <= t_s < r.getCurrentTime() (Note this could collapse to an
+         *    empty interval is r.getState() != 0)
+         *
+         * After successful f.wait_for or f.wait_until, the result is the same
+         * as above. If they time out, no observable state changes.
+         *
+         * If the state is FAILED when read is called the result is undefined.
+         *
+         * @param min The minimum end time until which spikes will be read. If
+         *        UNDEFINED_TIMESTAMP, it will be considered as infinite.
+         */
+        boost::future< Spikes > read( float min );
+
+        /**
+         * Read all available spikes.
+         *
+         * This is a convenience function that behaves like:
+         * read( -inf ) is getCurrentTime() is undefined and
+         * read( getCurrentTime( )) otherwise.
+         */
+        boost::future< Spikes > read();
+
+        enum StateFlags { ENDED = 1, FAILED = 2 };
```

Comment:
It would be three states then. Whatever you prefer, I don't have a strong opinion about this.